Mathlib/FieldTheory/Galois/NormalBasis.lean

```diff
@@ -0,0 +1,127 @@
+/-
+Copyright (c) 2025 Madison Crim, Aaron Liu, Justus Springer, Junyan Xu. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Madison Crim, Aaron Liu, Justus Springer, Junyan Xu
+-/
+import Mathlib.Algebra.Module.PID
+import Mathlib.Algebra.MvPolynomial.Funext
+import Mathlib.Algebra.Polynomial.Module.AEval
+import Mathlib.FieldTheory.Finite.Basic
+import Mathlib.FieldTheory.Galois.Basic
+import Mathlib.LinearAlgebra.AnnihilatingPolynomial
+import Mathlib.LinearAlgebra.Matrix.Nondegenerate
+
+/-!
+# The normal basis theorem
+
+We prove the normal basis theorem `IsGalois.normalBasis`:
+every finite Galois extension has a basis that is an orbit under the Galois group action.
+
+The proof follows [ConradLinearChar] Keith Conrad, *Linear Independence of Characters*.
+
+-/
+
+variable (K L : Type*) [Field K] [Field L] [Algebra K L]
+
+open Polynomial FiniteField Module Submodule LinearMap in
+-- [ConradLinearChar] Theorem 3.7.
+private theorem exists_linearIndependent_algEquiv_apply_of_finite [Finite L] :
+    ∃ x : L, LinearIndependent K fun σ : L ≃ₐ[K] L ↦ σ x := by
+  have := Finite.of_injective _ (algebraMap K L).injective
+  have := Fintype.ofFinite K
+  /- Since `K[X]` is a PID and `L` is a finitely generated `K[X]`-module (with `X` acting as
+    the Frobenius automorphism), there exists `x : L` whose `K[X]`-annihilator is equal to the
+    `K[X]`-annihilator of `L`, which is generated by `X ^ [L : K] - 1`. -/
+  have ⟨x, hx⟩ := exists_ker_toSpanSingleton_eq_annihilator K[X]
+    (AEval' (frobeniusAlgHom K L).toLinearMap)
```

Comment:
```suggestion
  obtain ⟨x, hx⟩ := exists_ker_toSpanSingleton_eq_annihilator K[X]
    (AEval' (frobeniusAlgHom K L).toLinearMap)
```

Reply:
This probably simplifies infoview a bit (didn't check) so I'll take it.